### PR TITLE
mkdown conversion stuff

### PR DESCRIPTION
**Pull request by Chrisvann1**
This includes the mkdown conversion functionality for both the frontend and the backend. On the backend end of things, it creates a subdirectory called "mkDown_dir" if it does not already exist. The backend will convert the content of notes that the user wants to convert and store them inside of this subdirectory. As for the frontend, it takes user input and uses an HTTP request containing the title and the content that will be converted. I already tested it and it seems to work.

### CODE REVIEW

File: frontend.py [hunk 1]
```diff
@@ -79,7 +79,8 @@ def printAppUse():
 	printColor("2. Edit Note(Limited Functionality in v0.01)", getConfig(2),"")
 	printColor("3. Search Notes",getConfig(2),"")
 	printColor("4. List Notes",getConfig(2),"")
-	printColor("5. Delete Note",getConfig(2))
+	printColor("5. Delete Note",getConfig(2),"")
+	printColor("6. Convert Note to MKDown",getConfig(2))
```
Comment (augustus-thomas) [inline frontend.py:83]: these two lines are conflicting with the frontend.py in main and they cause a merge conflict. **What is the difference between converting a note to MKDown and exporting to PDF**? Option 6 in the current version asks for exporting to PDF, here you have converting to markdown. Do we want 7 options, or do we just want you convert to MKDown option? I am not sure when exporting to PDF was added.

File: apiCalls.py [hunk 2]
```diff
@@ -55,6 +55,10 @@ def addContent(noteTitle, noteContent):
 	url = "http://127.0.0.1:5000/notes"
 	return requests.put(url,headers=header, json={'title': noteTitle, 'content': noteContent}) 
 
+def mdDownConversion(noteTitle, noteContent): 
+    url = "http://127.0.0.1:5000/mkdown"
+    return requests.post(url, headers=header, json={'title': noteTitle, 'content': noteContent})
```
Comment (augustus-thomas) [inline apiCalls.py:60]: alright so noteTitle and noteContent need to be cast as strings by the caller.

File: app.py [hunk 3]
```diff
@@ -410,5 +417,26 @@ def list_tags():
...
+        abort(206, error)
+        
+    try: 
+        content = payload.get('content')
+        if content is None: 
+            raise KeyError("Data Conversion Requires content")
+    except KeyError as error: 
+        abort(206, error)
+        
+    with open("./MKDown_dir/" + title, "w") as mk:
```
Comment (augustus-thomas) [inline app.py:438]: no .md file extension on a markdown file?

File: frontend.py [hunk 4]
```diff
@@ -399,7 +400,26 @@ def runtime(state):
...
+				printColor("What is the title of the note you would like to convert to MKDown?",getConfig(2), "")
+				lineBreak(columns,getConfig(4))
+				search_by = input(": ")
+				desired_response = ['content']
+				api_response = apiCalls.searchNotes('title', search_by, desired_response)
+				entries = translation(api_response.content)
+				parsing = entries.split("\n")
+				content = parsing[1].replace(r'\n', '\n')
+				title = parsing[0]
+				if ".mk" not in title:
```
Comment (augustus-thomas) [inline frontend.py:415]: markdown files have .md file extensions not .mk

File: app.py (hunk 3 above)
Comment (augustus-thomas) [reply app.py:438]: nevermind i see that this is being enforced by the frontend caller.

**Review by augustus-thomas — CHANGES_REQUESTED** [review]
please see my nodes on markdown file extensions and on the merge conflict with the export to PDF option on the frontend

**Comment by augustus-thomas** [comment]
Running `git blame` on the merge conflict finds that it's from commit [19f3a395](https://github.com/Chrisvann1/Software-Engineering-Note-Taking-Application/commit/19f3a395ef2b73d478ceadb8c01de163ff97955e) by @jteferi yesterday. There was no PR for that, it was pushed directly to main. Maybe we can discuss a design to resolve this?

File: frontend.py (hunk 4 above)
Comment (Chrisvann1) [reply frontend.py]: I fixed this

File: frontend.py (hunk 1 above)
Comment (Chrisvann1) [reply frontend.py]: I changed mine to option 7 so it would be easier to deal with merge conflicts.

**Pull request merged**